dev/Repeater/ElementManager.cpp

```diff
@@ -270,8 +270,36 @@ void ElementManager::DataSourceChanged(const winrt::IInspectable& /*source*/, wi
 
         case winrt::NotifyCollectionChangedAction::Replace:
         {
-            OnItemsRemoved(args.OldStartingIndex(), args.OldItems().Size());
-            OnItemsAdded(args.NewStartingIndex(), args.NewItems().Size());
+            int oldSize = args.OldItems().Size();
+            int newSize = args.NewItems().Size();
+            int oldStartIndex = args.OldStartingIndex();
+            int newStartIndex = args.NewStartingIndex();
+
+            if (oldSize == newSize &&
+                oldStartIndex == newStartIndex &&
+                IsDataIndexRealized(oldStartIndex) &&
+                IsDataIndexRealized(oldStartIndex + oldSize))
+            {
+                // Straight up replace of n items within the realization window.
+                // Removing and adding might causes us to lose the anchor causing us
+                // to throw away all containers and start from scratch.
+                // Instead, we can just clear those items and set the element to
+                // null (sentinel) and let the next measure get new containers for them.
+                auto startRealizedIndex = GetRealizedRangeIndexFromDataIndex(oldStartIndex);
```

Comment:
This code block looks a lot like ClearRealizedRange. Can we reuse that function?

Reply:
I thought about that initially but there is a subtle difference. Clear just gets rid of the items and updates the realized range in element manager. Here we just want to clear the elements and fill with sentinels. It is as if nothing happened to the realized range.

Reply:
Turns out there are currently no collections that use more complex ranges for replace, so leaving the change as is and we can make it more complex when necessary :)